wondrous-app/components/Common/Votes/index.tsx

```diff
@@ -1,94 +1,149 @@
-  VoteButton,
+  VoterProfilePicture,
+  VoteLabel,
   VoteRowWrapper,
   VoteProgressBar,
   VoteCurrentProgress,
-  VoteRowResult,
-  VotePercentageResult,
-  RetractButton,
-  VoteButtonLabel,
+  VoteRowContentWrapper,
 } from './styles';
 
+const computePercentage = (votes, type) => {
+  const amount = votes?.counts[type] || 0;
+  return Math.round((100 * amount) / votes.totalVotes) || 0;
+};
+
 interface Props {
-  votes: {
-    counts: {
-      approve: number;
-      reject: number;
+  fullScreen: boolean;
+  proposal: {
+    id: string;
+    orgId: string;
+    podId: string;
+    org: {
+      privacyLevel: string;
+    };
+    pod: {
+      privacyLevel: string;
     };
-    userVote: string;
+    votes: {
+      counts: Record<string, number>;
+      userVote: string;
+      totalVotes: number;
+    };
+    voteOptions: string[];
+    voteType: 'binary' | 'custom';
   };
-  fullScreen: boolean;
-  taskId: string;
-  totalVotes: number;
   proposalStatus: string;
+  userInOrg: boolean;
+}
+function VoteOptionsRow({ votes, option, handleVote, handleUndoVote, user, voteType }) {
+  const percentage = computePercentage(votes, option);
+  const userVotedFor = votes?.userVote === option;
+  let optionDisplayName = option;
+  if (voteType === 'binary') {
+    optionDisplayName = option === 'approve' ? 'Yes' : 'No';
+  }
+  return (
+    <VoteRowWrapper key={`${option}`}>
+      <VoteRowContentWrapper onClick={() => handleVote(option)}>
+        <div style={{ display: 'flex', gap: 5 }}>
+          <VoteLabel color={palette.blue20}>{percentage}%</VoteLabel>
+          <VoteLabel color={palette.white} weight={500}>
+            {optionDisplayName}
+          </VoteLabel>
+        </div>
+        <div style={{ display: 'flex', gap: 5 }}>
+          {userVotedFor ? (
+            <>
+              <VoteLabel
+                style={{ cursor: 'pointer' }}
+                color={palette.highlightBlue}
+                weight={500}
+                onClick={(e) => handleUndoVote(e)}
+              >
+                Undo vote
+              </VoteLabel>
+              <VoterProfilePicture src={user?.thumbnailPicture || user?.profilePicture} />
+              <VotedForCheckMark />{' '}
+            </>
+          ) : (
+            <NotVotedCheckmark />
+          )}
+        </div>
+      </VoteRowContentWrapper>
+      <VoteProgressBar>
+        <VoteCurrentProgress color="linear-gradient(90deg, #4F00DE 65%, #06FFA5 100%)" width={`${percentage}%`} />
+      </VoteProgressBar>
+    </VoteRowWrapper>
+  );
 }
 
-export default function VoteResults({ votes, fullScreen, taskId, totalVotes, proposalStatus }: Props) {
+export default function VoteResults({ userInOrg, proposal, fullScreen, proposalStatus }: Props) {
+  const { votes } = proposal;
+  const { totalVotes } = votes;
+  const proposalId = proposal.id;
   const user = useMe();
+
+  const canVote = user && userInOrg && proposalStatus === STATUS_OPEN; // TODO add logic for private pod proposals?
```

Comment:
It shouldn't be visible in private pod right?

Reply:
Yeah logic a bit complicated I was gonna add it later